### ISSUE TITLE
Fix failed unit tests after 2.2 version bump

### ISSUE THREAD

**Issue by ohltyler** [opensearch-project/anomaly-detection#630]
After version bump to 2.2, seeing many failures:
```
./gradlew build -Dtest.logs=true

...

1256 tests completed, 92 failed, 9 skipped
```

**Comment by ohltyler** [comment]
Almost all tests related to `SearchFeatureDaoTests`. Common error is issues finding `ClusterSettings`:
```
java.lang.NoClassDefFoundError: Could not initialize class org.opensearch.common.settings.ClusterSettings
```

**Comment by ohltyler** [comment]
And failure to initialize:
```
java.lang.ExceptionInInitializerError
	at org.opensearch.common.settings.ClusterSettings.<clinit>(ClusterSettings.java:219)
```

**Comment by amitgalitz** [comment]
See a similar thing in release notes PR:
https://github.com/opensearch-project/anomaly-detection/runs/7695563475?check_suite_focus=true
```
1256 tests completed, 92 failed, 9 skipped
```
Most tests are either in `SearchFeatureDaoTests` or `ModelManagerTests`,

Also seeing the same failure to initialize `ClusterSettings`

**Comment by ohltyler** [comment]
Some findings:

These are the only test classes that are both (1) not extending a test class that is extending `OpenSearchTestCase` (directly or indirectly) and (2) defining `ClusterSettings` var in `setup()`

All tests were passing until the version bump to 2.2

We've tried forcing `OpenSearchTestCase` to be the same as 2.1, by way of forcing the lucene test framework (which is the base package used for `OpenSearchTestCase`) as the version used in 2.1

These 2 test classes are annotated with `@RunWith(PowerMockRunner.class)`. There is one other class that also does this (`GetAnomalyDetectorActionTests`), but it does not define a `ClusterSettings`. The tests in this class pass.

**Comment by ohltyler** [comment]
Update: after adding `@RunWith(PowerMockRunner.class)` to `GetAnomalyDetectorActionTests` and then trying to initialize `ClusterSettings` var, causes it to fail with similar errors. So it is related to the runner not being able to pull in the dependencies correctly somehow.

**Comment by ohltyler** [comment]
Resolved in #637
